crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -117,6 +122,7 @@ module sui::sui_system {
         stake: Coin<SUI>,
         gas_price: u64,
         commission_rate: u64,
+        chain_id: vector<u8>,
```

Comment:
The benefit of using `TransactionDigest::genesis()` as the chain ID is that we do not need to include the chain ID here or (in any other message that's part of a tx) because every tx implicitly commits to it.

Reply:
In more detail (since I realize the above is a bit terse).
- Let's say the system object on testnet has id `A` and the system object on mainnet has id `B`.
- We know (modulo hash collisions) that `A != B`, and (even stronger) `A` is not even a valid object ID on mainnet, and `B` is not a valid object ID on testnet. This is because all object ID's (indirectly) contain a commitment to the genesis tx hash.
-  Thus, if the (e.g.) sender directs a tx intended for testnet to mainnet, it will fail long before reaching line 137 below. All validators should refuse to sign the tx because they will be unable to reslve `A` to an object on mainnet, and thus a cert will never be formed, and the code below will not get executed.

Reply:
that makes sense for a user transaction that the object used should not resolved to the object on the chain, hence replay tx will not work. 

but this part is for `request_add_validator`, this is not necessary either? is it because the object (i.e. stake) cannot be valid on a different chain for the same reason? 

in that case, we just need to add the chain_Id field to SuiSystemObject and allow an endpoint to query this field, and no need to check this when adding validator

Reply:
> but this part is for request_add_validator, this is not necessary either? is it because the object (i.e. stake) cannot be valid on a different chain for the same reason?

Correct

> in that case, we just need to add the chain_Id field to SuiSystemObject and allow an endpoint to query this field, and no need to check this when adding validator

Exactly, that makes sense to me.